### PR TITLE
fix: generate correct URLs for root bots

### PR DESCRIPTION
**Pull request by beyackle**
## Description

This fixes a small glitch in generating URLs for the links; dialogs and triggers that belong to the root of a botProject now generate links that look like the older non-botProject ones, which should fix some issues with undo/redo.

## Task Item

#minor

## Screenshots

image

### CODE REVIEW

**Review by cwhitten — APPROVED** [review]

File: Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx [hunk 1]
```diff
@@ -279,7 +279,7 @@ export const ProjectTree: React.FC<Props> = ({
       displayName: dialog.displayName,
       isRoot: dialog.isRoot,
       projectId: rootProjectId,
-      skillId: skillId,
+      skillId: skillId === rootProjectId ? undefined : skillId,
```
Comment (srinaath) [inline Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx:282]: @beyackle  I see a bug where I click on a skill's dialog file url changes correctly to `http://localhost:3000/bot/25141.842060596977/skill/5927.301138798824/dialogs/googlekeepsync?selected=triggers[%22376720%22]` . Navigate to a different page and come back to the DesignPage. The url is an invalid URL. It needs to set the url obtained from the previous designPageLocationState right?

**Review by srinaath — CHANGES_REQUESTED** [review]

File: Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx (hunk 1 above)
Comment (beyackle) [reply Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx:282]: This looks to me like a problem in the sidebar - when I have a skill open (so, the URL is /bot/<bot Id>/skill/<skill Id>) and click on the sidebar button, it takes me to a URL with /bot/<skill Id>/ in it, and then clicking back into Design keeps that around. Single-bot projects don't have this problem. The reason this change is going into main and not the feature branch is that undo/redo was breaking even in the single-bot case, right?

Comment (srinaath) [reply Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx:282]: This fix has no relation to undo/redo. The URL was broken irrespective of whether we were doing Undo/redo.

Comment (srinaath) [reply Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx:282]: This is also not an issue with the sidebar. DesignPageLocation state stores the current focussed actions etc that are particular to the design page. On rendering the designPage it should always use the currentProjectId state and parse the designPagelocation state to set the URL correctly?

**Review by srinaath — APPROVED** [review]

**Pull request merged**